<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>An introduction to the isajsonr package • isajsonr</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to the isajsonr package">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">isajsonr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html" aria-label="home"><span class="fa fa-home"></span></a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/index.html">Vignettes</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="nav-link" href="../news/index.html"><span class="fa fa-newspaper-o"></span> news</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-file-code-o"></span> functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-other-packages" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-seedling"></span> other packages</button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-other-packages">
<li><a class="external-link dropdown-item" href="https://biometris.github.io/isatabr">isatabr</a></li>
    <li><a class="external-link dropdown-item" href="https://biometris.github.io/statgenSTA">statgenSTA</a></li>
    <li><a class="external-link dropdown-item" href="https://biometris.github.io/statgenGxE">statgenGxE</a></li>
    <li><a class="external-link dropdown-item" href="https://biometris.github.io/statgenGWAS">statgenGWAS</a></li>
    <li><a class="external-link dropdown-item" href="https://biometris.github.io/statgenHTP">statgenHTP</a></li>
    <li><a class="external-link dropdown-item" href="https://biometris.github.io/statgenIBD">statgenIBD</a></li>
    <li><a class="external-link dropdown-item" href="https://biometris.github.io/statgenMPP">statgenMPP</a></li>
    <li><a class="external-link dropdown-item" href="https://biometris.github.io/LMMsolver">LMMsolver</a></li>
  </ul>
</li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Biometris/isajsonr"><span class="fa fa-github fa-lg fab"></span> github</a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>An introduction to the isajsonr package</h1>
                        <h4 data-toc-skip class="author">Bart-Jan van
Rossum</h4>
            
            <h4 data-toc-skip class="date">2024-11-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/Biometris/isajsonr/blob/main/vignettes/isajsonr.Rmd" class="external-link"><code>vignettes/isajsonr.Rmd</code></a></small>
      <div class="d-none name"><code>isajsonr.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 class="unnumbered" id="the-isajsonr-package">The isajsonr package<a class="anchor" aria-label="anchor" href="#the-isajsonr-package"></a>
</h2>
<p>The isajsonr package is developed as a easy-to-use package for
reading, modifying and writing files in the Investigation/Study/Assay
(ISA) Abstract Model of the metadata framework using the JSON
format.</p>
<p>ISA is a metadata framework to manage an increasingly diverse set of
life science, environmental and biomedical experiments that employ one
or a combination of technologies. Built around the
<strong>Investigation</strong> (the project context),
<strong>Study</strong> (a unit of research) and <strong>Assay</strong>
(analytical measurements) concepts, ISA helps you to provide rich
descriptions of experimental metadata (i.e. sample characteristics,
technology and measurement types, sample-to-data relationships) so that
the resulting data and discoveries are reproducible and reusable.</p>
</div>
<div class="section level2">
<h2 id="the-isa-json-structure">The ISA-JSON structure<a class="anchor" aria-label="anchor" href="#the-isa-json-structure"></a>
</h2>
<p>The ISA-JSON format is described in full detail on <a href="https://isa-specs.readthedocs.io/en/latest/isajson.html" class="external-link uri">the ISA-JSON website</a>.</p>
<p>All ISA-JSON content regarding multiple Study and Assay should fall
under one Investigation JSON structure, therefore should be recorded in
a single JSON file. The JSON file SHOULD have a .json extension.</p>
<div class="section level3">
<h3 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>As an example for working with the <code>isajsonr</code> package we
will use the data set that accompanies <span class="citation">Castrillo
et al. (2007)</span>. The associated file is included in the
package.</p>
</div>
</div>
<div class="section level2">
<h2 id="reading-files-in-the-isa-json-format">Reading files in the ISA-JSON format<a class="anchor" aria-label="anchor" href="#reading-files-in-the-isa-json-format"></a>
</h2>
<p>ISA-JSON files can be stored in two different ways, either as
stand-alone file in a directory, or as .zip file containing the file.
The example data is included as stand-alone file in the package. Both
formats can be read into <code>R</code> using the
<code>readISAJSON</code> function.</p>
<p>When reading an ISA-JSON file from a directory, the full location of
the file has to be specified. The package will recognize a zipped file
and extract it into a temporary folder before reading it.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Read ISA-JSON file from directory.</span></span>
<span><span class="va">isaObject1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readISAJSON.html">readISAJSON</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/Castrillo/Castrillo.json"</span>, </span>
<span>                                                       package <span class="op">=</span> <span class="st">"isajsonr"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>In both cases <code>readISATab</code> will validate the content of
the .json file against the isa-json model scheme which can be found <a href="https://github.com/ISA-tools/isa-api/tree/master/isatools/resources/schemas/isa_model_version_1_0_schemas/core" class="external-link">here</a>.
The imported ISA-JSON file is stored in an object of the S4 class
<code>ISAjson</code>. Since the information is almost identical for
reading content from an unzipped and a zipped-file, the following
sections will show the example for the files read from an unzipped file
only.</p>
</div>
<div class="section level2">
<h2 id="accessing-and-updating-isa-objects-">Accessing and updating ISA objects.<a class="anchor" aria-label="anchor" href="#accessing-and-updating-isa-objects-"></a>
</h2>
<p>All information from the ISA-JSON file is stored within the
<code>content</code> slot in the <code>ISAjson</code> object. The path
where the file was originally read from is stored in the
<code>path</code> slot.</p>
<p>All section in the ISA structure have corresponding functions for
accessing and modifying information. The names of these access functions
correspond to the section they refer to, e.g. accessing the Ontology
Source Reference (OSR) section in an <code>ISAjson</code> object can be
done using the <code><a href="../reference/oSR.html">oSR()</a></code> function. There is one notable
exception to this. To prevent problems with the <code>path()</code>
function, that already exists in quite some other packages, the path
slot in an <code>ISA</code> object should be accessed using the
<code><a href="../reference/isaPath.html">isaPath()</a></code> function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Access path for isaObjects</span></span>
<span><span class="fu"><a href="../reference/isaPath.html">isaPath</a></span><span class="op">(</span><span class="va">isaObject1</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/home/runner/work/_temp/Library/isajsonr/extdata/Castrillo/Castrillo.json"</span></span></code></pre></div>
<p>The path for <code>isaObject1</code> shows the full path to the file
that was read using <code>readISAJSON</code>.</p>
<p>The other sections are accessible in a similar way. Some more
examples are shown below.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Access studies.</span></span>
<span><span class="va">isaStudies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/study.html">study</a></span><span class="op">(</span><span class="va">isaObject1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Print study names.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">isaStudies</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "filename"                      "identifier"                    "title"                        </span></span>
<span><span class="co">#&gt; [4] "description"                   "submissionDate"                "publicReleaseDate"            </span></span>
<span><span class="co">#&gt; [7] "Comment[Study Funding Agency]" "Comment[Study Grant Number]"</span></span>
<span></span>
<span><span class="co">## Access study descriptors.</span></span>
<span><span class="va">isaSDD</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sDD.html">sDD</a></span><span class="op">(</span><span class="va">isaObject1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Shows study descriptor for study s_BII-S-1.</span></span>
<span><span class="va">isaSDD</span><span class="op">$</span><span class="va">`s_BII-S-1.txt`</span></span>
<span><span class="co">#&gt;       annotationValue termSource                              termAccession</span></span>
<span><span class="co">#&gt; 1 intervention design        OBI http://purl.obolibrary.org/obo/OBI_0000115</span></span></code></pre></div>
<p>It is not only possible to access the information in the different
sections in an <code>ISAjson</code> object, the information can also be
updated. As the access functions, the update functions have the same
name as the sections they refer to. As an example, let’s assume an error
sneaked into the OSR section and we want to update one of the source
versions.</p>
<p>First have a look at the current content of the OSR section.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">isaOSR</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/oSR.html">oSR</a></span><span class="op">(</span><span class="va">isaObject1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                 description                                             file  name</span></span>
<span><span class="co">#&gt; 1    Ontology for Biomedical Investigations http://bioportal.bioontology.org/ontologies/1123   OBI</span></span>
<span><span class="co">#&gt; 2             BRENDA tissue / enzyme source        ArrayExpress Experimental Factor Ontology   BTO</span></span>
<span><span class="co">#&gt; 3            NEWT UniProt Taxonomy Database                                                   NEWT</span></span>
<span><span class="co">#&gt; 4                             Unit Ontology                                                     UO</span></span>
<span><span class="co">#&gt; 5  Chemical Entities of Biological Interest                                                  CHEBI</span></span>
<span><span class="co">#&gt; 6         Phenotypic qualities (properties)                                                   PATO</span></span>
<span><span class="co">#&gt; 7 ArrayExpress Experimental Factor Ontology                                                    EFO</span></span>
<span><span class="co">#&gt;   version</span></span>
<span><span class="co">#&gt; 1   47893</span></span>
<span><span class="co">#&gt; 2  v 1.26</span></span>
<span><span class="co">#&gt; 3  v 1.26</span></span>
<span><span class="co">#&gt; 4  v 1.26</span></span>
<span><span class="co">#&gt; 5  v 1.26</span></span>
<span><span class="co">#&gt; 6  v 1.26</span></span>
<span><span class="co">#&gt; 7  v 1.26</span></span></code></pre></div>
<p>Now we update the version of the Ontology for Biomedical
Investigations from 47893 to 47894. Then we update the modified ontology
source <code>data.frame</code> in the <code>ISAjson</code> object.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Update version number.</span></span>
<span><span class="va">isaOSR</span><span class="op">[</span><span class="fl">1</span>, <span class="st">"version"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">47894</span></span>
<span></span>
<span><span class="co">## Update oSR in ISAjson object.</span></span>
<span><span class="fu"><a href="../reference/oSR.html">oSR</a></span><span class="op">(</span><span class="va">isaObject1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">isaOSR</span></span>
<span></span>
<span><span class="co">## Check the updated oSR.</span></span>
<span><span class="fu"><a href="../reference/oSR.html">oSR</a></span><span class="op">(</span><span class="va">isaObject1</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                 description                                             file  name</span></span>
<span><span class="co">#&gt; 1    Ontology for Biomedical Investigations http://bioportal.bioontology.org/ontologies/1123   OBI</span></span>
<span><span class="co">#&gt; 2             BRENDA tissue / enzyme source        ArrayExpress Experimental Factor Ontology   BTO</span></span>
<span><span class="co">#&gt; 3            NEWT UniProt Taxonomy Database                                                   NEWT</span></span>
<span><span class="co">#&gt; 4                             Unit Ontology                                                     UO</span></span>
<span><span class="co">#&gt; 5  Chemical Entities of Biological Interest                                                  CHEBI</span></span>
<span><span class="co">#&gt; 6         Phenotypic qualities (properties)                                                   PATO</span></span>
<span><span class="co">#&gt; 7 ArrayExpress Experimental Factor Ontology                                                    EFO</span></span>
<span><span class="co">#&gt;   version</span></span>
<span><span class="co">#&gt; 1   47894</span></span>
<span><span class="co">#&gt; 2  v 1.26</span></span>
<span><span class="co">#&gt; 3  v 1.26</span></span>
<span><span class="co">#&gt; 4  v 1.26</span></span>
<span><span class="co">#&gt; 5  v 1.26</span></span>
<span><span class="co">#&gt; 6  v 1.26</span></span>
<span><span class="co">#&gt; 7  v 1.26</span></span></code></pre></div>
<p>In a similar way all sections in an <code>ISAjson</code> object can
be accessed and updated.</p>
</div>
<div class="section level2">
<h2 id="processing-assay-section">Processing assay section<a class="anchor" aria-label="anchor" href="#processing-assay-section"></a>
</h2>
<p>The assay section may contain information about the files used to
store the actual data for the assay. Per assay section two types of data
files may be referred to: 1) the file(s) containing the raw data, and 2)
the file(s) containing derived data.</p>
<p>Looking at the assay section in our example data, we see that the
<code>a_proteome</code> assay in survey <code>s_BII-S-1</code> only
contains both raw and derived data files, as shown in the dataFileType
column in the output.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Inspect assay tab for survey s_BII-S-1.</span></span>
<span><span class="va">isaAFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/aFiles.html">aFiles</a></span><span class="op">(</span><span class="va">isaObject1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">isaAFile</span><span class="op">$</span><span class="va">`s_BII-S-1.txt`</span><span class="op">$</span><span class="va">a_proteome.txt</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                                    dataFile@id                   dataFilename</span></span>
<span><span class="co">#&gt; 1                     #data/proteinassignmentfile-proteins.csv                   proteins.csv</span></span>
<span><span class="co">#&gt; 2 #data/derivedspectraldatafile-PRIDE_Exp_Complete_Ac_8763.xml PRIDE_Exp_Complete_Ac_8763.xml</span></span>
<span><span class="co">#&gt; 3                    #data/rawspectraldatafile-spectrum.mzdata                spectrum.mzdata</span></span>
<span><span class="co">#&gt; 4   #data/posttranslationalmodificationassignmentfile-ptms.csv                       ptms.csv</span></span>
<span><span class="co">#&gt; 5 #data/derivedspectraldatafile-PRIDE_Exp_Complete_Ac_8762.xml PRIDE_Exp_Complete_Ac_8762.xml</span></span>
<span><span class="co">#&gt; 6 #data/derivedspectraldatafile-PRIDE_Exp_Complete_Ac_8761.xml PRIDE_Exp_Complete_Ac_8761.xml</span></span>
<span><span class="co">#&gt;                                      dataFiletype dataFileComment[PRIDE Accession]</span></span>
<span><span class="co">#&gt; 1                         Protein Assignment File                             8761</span></span>
<span><span class="co">#&gt; 2                      Derived Spectral Data File                             8763</span></span>
<span><span class="co">#&gt; 3                          Raw Spectral Data File                             8761</span></span>
<span><span class="co">#&gt; 4 Post Translational Modification Assignment File                             8761</span></span>
<span><span class="co">#&gt; 5                      Derived Spectral Data File                             8762</span></span>
<span><span class="co">#&gt; 6                      Derived Spectral Data File                             8761</span></span>
<span><span class="co">#&gt;   dataFileComment[PRIDE Processed Data Accession]</span></span>
<span><span class="co">#&gt; 1                                            8761</span></span>
<span><span class="co">#&gt; 2                                            8763</span></span>
<span><span class="co">#&gt; 3                                            8761</span></span>
<span><span class="co">#&gt; 4                                            8761</span></span>
<span><span class="co">#&gt; 5                                            8762</span></span>
<span><span class="co">#&gt; 6                                            8761</span></span></code></pre></div>
<p>To read the contents of the data files, either raw or derived, in the
assay tab file, we can use the <code><a href="../reference/processAssay-methods.html">processAssay()</a></code> function. The
exact working of this function depends on the technology type of the
assay. For most technology types the data files are read as plain
<code>.txt</code> files assuming a tab-delimited format. Only for mass
spectrometry and microarray data the files are read differently (see the
sections below). The protein assignment file in the example above is a
regular data file and will be read as such.</p>
<p>Before being able to process the assay file, i.e. read the data, we
first have to extract the assay tabs using the
<code><a href="../reference/getAssayTabs.html">getAssayTabs()</a></code> function. This function extracts all the
assay files from an <code>ISAjson</code> object and stores them as
<code>assayTab</code> objects. These <code>assayTab</code> objects
contain not only the content of the assay tab file, but also extra
information, e.g. technology type.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get assay tabs for isaObject1.</span></span>
<span><span class="va">aTabObjects</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getAssayTabs.html">getAssayTabs</a></span><span class="op">(</span><span class="va">isaObject1</span><span class="op">)</span></span>
<span><span class="va">aTabObjects</span><span class="op">$</span><span class="va">`s_BII-S-1.txt`</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">aFile</span> <span class="op">&lt;-</span> <span class="va">aTabObjects</span><span class="op">$</span><span class="va">`s_BII-S-1.txt`</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">aFile</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/startsWith.html" class="external-link">startsWith</a></span><span class="op">(</span><span class="va">aTabObjects</span><span class="op">$</span><span class="va">`s_BII-S-1.txt`</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">aFile</span><span class="op">$</span><span class="va">dataFilename</span>, prefix <span class="op">=</span> <span class="st">"/"</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## Process assay data.</span></span>
<span><span class="va">isaDat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/processAssay-methods.html">processAssay</a></span><span class="op">(</span>isaObject <span class="op">=</span> <span class="va">isaObject1</span>,</span>
<span>                       aTabObject <span class="op">=</span> <span class="va">aTabObjects</span><span class="op">$</span><span class="va">`s_BII-S-1.txt`</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                       type <span class="op">=</span> <span class="st">"derived"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in bpstopOnError(BPPARAM) : could not find function "bpstopOnError"</span></span>
<span> </span>
<span><span class="co">## Display first rows and columns.</span></span>
<span><span class="co">#head(isaDat[, 1:10])</span></span></code></pre></div>
<p>The data is now stored in <code>isaDat</code> and can be used for
further analysis within <code>R</code>.</p>
<div class="section level3">
<h3 id="mass-spectrometry-assay-files">Mass spectrometry assay files<a class="anchor" aria-label="anchor" href="#mass-spectrometry-assay-files"></a>
</h3>
<p>Mass spectrometry data is often stored in Network Common Data Form
(NetCDF) files, i.e. in .CDF files. Assay data containing these data
will be processed in a different way than regular assay data. To be able
to do this the <a href="https://bioconductor.org/packages/release/bioc/html/xcms.html" class="external-link">xcms</a>
package is required. This package is available from <a href="https://bioconductor.org/" class="external-link">Bioconductor</a>.</p>
<p>After reading the ISA-Tab files, we can now process the mass
spectrometry assay data. In this example the raw data is available, so
when processing the assay we specify <code>type = "raw"</code>. The rest
of the code is similar to the previous section.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get assay tabs for isaObject3.</span></span>
<span><span class="co"># aTabObjects3 &lt;- getAssayTabs(isaObject3)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># ## Process assay data.</span></span>
<span><span class="co"># isaDat3 &lt;- processAssay(isaObject = isaObject3,</span></span>
<span><span class="co">#                         aTabObject = aTabObjects3$s_Proteomic_profiling_of_yeast.txt$a_metabolite.txt,</span></span>
<span><span class="co">#                         type = "raw")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># ## Display output.</span></span>
<span><span class="co"># isaDat3</span></span></code></pre></div>
<p>As the output shows, processing the mass spectrometry data gives an
object of class <code>xcmsSet</code> from the <code>xcms</code> package.
This object contains all available information from the .CDF file that
was read and can be used for further analysis.</p>
</div>
<div class="section level3">
<h3 id="microarray-assay-files">Microarray assay files<a class="anchor" aria-label="anchor" href="#microarray-assay-files"></a>
</h3>
<p>Microarray data is often stored in an Affymetrix Probe Results file.
These <code>.CEL</code> files contain information on the probe set’s
intensity values, and a probe set represents a gene. Assay data
containing these data will be processed in a different way than regular
assay data. To be able to do this the <a href="https://www.bioconductor.org/packages/release/bioc/html/affy.html" class="external-link">affy</a>
package is required. This package is available from <a href="https://bioconductor.org/" class="external-link">Bioconductor</a>.</p>
<p>Processing microarray data is done in a very similar way as
processing mass spectrometry data, as described in the previous section.
The main difference is that the resulting object will in this case be an
object object of class <code>ExpressionSet</code>, which is used as
input in many Bioconductor packages.</p>
</div>
</div>
<div class="section level2">
<h2 id="writing-files-in-the-isa-json-format-">Writing files in the ISA-JSON format.<a class="anchor" aria-label="anchor" href="#writing-files-in-the-isa-json-format-"></a>
</h2>
<p>After updating an <code>ISAjson</code> object, it can be written back
to a directory using the <code><a href="../reference/writeISAjson.html">writeISAjson()</a></code> function. All
content of the <code>ISAjson</code> object will be written to a json
file following the ISA-JSON standard specifications. The location of the
output file is specified in the <code>file</code> argument.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Write content of ISA object to a temporary file.</span></span>
<span><span class="fu"><a href="../reference/writeISAjson.html">writeISAjson</a></span><span class="op">(</span>isaObject <span class="op">=</span> <span class="va">isaObject1</span>, </span>
<span>             file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<hr>
<div class="section level3">
<h3 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Castrillo2007" class="csl-entry">
Castrillo, Juan I, Leo A Zeef, David C Hoyle, Nianshu Zhang, Andrew
Hayes, David CJ Gardner, Michael J Cornell, et al. 2007. <span>“Growth
Control of the Eukaryote Cell: A Systems Biology Study in Yeast.”</span>
<em>Journal of Biology</em> 6 (2). <a href="https://doi.org/10.1186/jbiol54" class="external-link">https://doi.org/10.1186/jbiol54</a>.
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bart-Jan van Rossum.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
